haskell/test/Stablecoin.hs

```diff
@@ -0,0 +1,107 @@
+-- SPDX-FileCopyrightText: 2020 tqtezos
+-- SPDX-License-Identifier: MIT
+
+module Stablecoin
+  ( spec_Stablecoin
+  , spec_StablecoinNettest
+  , stablecoinNettestReal
+  ) where
+
+import Test.Hspec
+
+import Tezos.Core
+import Data.Maybe
+import qualified Data.Map as Map
+import Lorentz.Address as LA
+import qualified Morley.Nettest as NetTest
+import Michelson.Test.Import
+import Michelson.Test.Integrational
+import Michelson.Typed
+import qualified Michelson.Untyped as U
+import qualified Lorentz as L
+import Lorentz.Contracts.Test.FA2
+import Lorentz.Contracts.Spec.FA2Interface hiding (Parameter)
+import Lorentz.Contracts.Stablecoin as SC
+import Nettest
+
+spec_Stablecoin :: Spec
+spec_Stablecoin = specWithUntypedContract "test/resources/stablecoin.tz" specCallback
+
+spec_StablecoinNettest :: Spec
+spec_StablecoinNettest = specWithUntypedContract "test/resources/stablecoin.tz" nettest_scenario
+
+stablecoinNettestReal :: Spec
+stablecoinNettestReal =
+  specWithUntypedContract "test/resources/stablecoin.tz" $ real_nettest_StablecoinNettest
+
+real_nettest_StablecoinNettest :: U.Contract ->  Spec
+real_nettest_StablecoinNettest contract = describe "Stablecoin real network behavior" $
+  it "can be originated and execute a transfer call" $ do
+    env <- NetTest.mkMorleyClientEnv $
+      NetTest.MorleyClientConfig
+        (Just "stablecoin_") Nothing Nothing "tezos-client" False True Nothing
+    NetTest.runNettestClient env (scTransferScenario (fromJust . mkInitialStorage) contract)
+
+nettest_scenario :: U.Contract  -> Spec
+nettest_scenario contract = describe "Stablecount nettest (integrational)" $
+  it "can be originated and execute a transfer call" $
+    integrationalTestExpectation $
+    NetTest.nettestToIntegrational $
+    scTransferScenario (fromJust . mkInitialStorage) contract
+
+originationRequestCompatible :: OriginationParams -> Bool
```

Comment:
What is this function for? Can you elaborate on why other permissions are not compatible for origination? And why do you restrict receiver and sender hooks? I guess there should be some documentation.

Reply:
There is some documentation here [1]. There is some discussion about this here as well [2]

But i ll add something here too.

[1] https://github.com/tqtezos/stablecoin/pull/34/files#diff-247038a0d8ab37be5403fa80adf5390aR130
[2] https://gitlab.com/morley-framework/morley/-/merge_requests/363#note_336764521

Reply:
Added.